src/components/export.tsx

```diff
@@ -1,51 +1,214 @@
 import Tooltip from 'react-bootstrap/Tooltip';
+
 import { MESSAGE_MAP } from '@isrd-isi-edu/chaise/src/utils/message-map';
+import { ConfigService } from '@isrd-isi-edu/chaise/src/services/config';
+import { LogService } from '@isrd-isi-edu/chaise/src/services/log';
+import { LogActions } from '@isrd-isi-edu/chaise/src/models/log';
+import ExportModal from '@isrd-isi-edu/chaise/src/components/export-modal';
+import useError from '@isrd-isi-edu/chaise/src/hooks/error';
+import { ChaiseAlertType } from '@isrd-isi-edu/chaise/src/providers/alerts';
+import useAlert from '@isrd-isi-edu/chaise/src/hooks/alerts';
 
 type ExportProps = {
-  reference: any,
-  disabled: boolean
-}
+  reference: any;
+  /**
+   * prop to make export button disable
+   */
+  disabled: boolean;
+};
 
+const Export = ({ reference, disabled }: ExportProps): JSX.Element => {
+  /**
+   * State variable to open progress modal dialog
+   */
+  const [showProgressModal, setShowProgressModal] = useState<boolean>(false);
+  /**
+   * State variable to export options
+   */
+  const [options, setOptions] = useState<Array>([]);
+  /**
+   * State Variable to store currently exporting object
+   */
+  const [selectedOption, setSelectedOption] = useState<any>(null);
+  /**
+   * State Variable to store exporter object which is used to cancel export.
+   */
+  const [exporterObj, setExporterObj] = useState<any>(null);
+  /**
+   * State variable to disable Export button 
+   */
+  const [isDownloading, setIsDownloading] = useState<boolean>(false);
+  /**
+   * State variable to control tooltip visiblity
+   */
+  const [showTooltip, setShowTooltip] = useState<boolean>(false);
 
-const Export = ({
-  reference,
-  disabled
-}: ExportProps): JSX.Element => {
+  const { dispatchError } = useError();
 
-  // TODO implement
+  const { addAlert } = useAlert();
 
+  useEffect(() => {
+    const options: any = [];
 
-  const [showSpinner, setShowSpinner] = useState(false);
+    if (reference) {
+      if (reference.csvDownloadLink) {
+        options.push({
+          displayname: 'Search results (CSV)',
+          type: 'DIRECT',
+        });
+      }
+  
+      const templates = reference.getExportTemplates(
+        !ConfigService.chaiseConfig.disableDefaultExport
+      );
+      
+      // Update the list of templates in UI
+      options.push(...templates);
+  
+      setOptions(options);
+    }
 
+  }, [reference]);
 
-  const renderExportIcon = () => {
-    if (showSpinner) return (<span className='chaise-btn-icon fa-solid fa-rotate fa-spin' />);
+  /**
+   * Send the request for export
+   * @param option selected export option
+   * @returns Downloads the selected file
+   */
+  const startExporting = (option: any) => () => {
+    const formatType = option.type;
+
+    switch(formatType) {
+      case 'DIRECT':
+        location.href = reference.csvDownloadLink;
+        break;
+      case 'BAG':
+      case 'FILE': 
+        setIsDownloading(true);
```

Comment:
Remember that using multipe set-states in a callback might cause batching issues. And to be honest you don't need this many state variables.. they all are doing the same thing.

Just set the `selectedOption`.  It makes the code easier to read as well.